Detectors/TPC/qc/src/PID.cxx

```diff
@@ -153,7 +171,20 @@ bool PID::processTrack(const o2::tpc::TrackTPC& track)
     if (std::abs(tgl) < mCutAbsTgl) {
       mMapHist["hdEdxTotVsp"][idEdxType]->Fill(pTPC, dEdxTot[idEdxType]);
       mMapHist["hdEdxMaxVsp"][idEdxType]->Fill(pTPC, dEdxMax[idEdxType]);
+      for (size_t iSpecies = 0; iSpecies < o2::track::PID::NIDs; ++iSpecies) {
+        dEdxRatio = dEdxTot[idEdxType] / mPID.getExpectedSignal(track, static_cast<o2::track::PID::ID>(iSpecies));
+      }
+      const auto PIDHypothesis = mPID.getMostProbablePID(track, mCutKrangeMin, mCutKrangeMax, mCutPrangeMin, mCutPrangeMax, mCutDrangeMin, mCutDrangeMax, mCutTrangeMin, mCutTrangeMax, 1, mPIDSigma);
+      int Hypothesis = static_cast<int>(PIDHypothesis) + 1;
+      if (static_cast<int>(PIDHypothesis) <= o2::track::PID::NIDs) {
+        if (track.getCharge() > 0) {
+          mMapHistCanvas["hdEdxVspHypoPos"][idEdxType]->SetBinContent(mMapHistCanvas["hdEdxVspHypoPos"][idEdxType]->GetXaxis()->FindBin(pTPC), mMapHistCanvas["hdEdxVspHypoPos"][idEdxType]->GetYaxis()->FindBin(dEdxTot[idEdxType]), Hypothesis);
+        } else {
+          mMapHistCanvas["hdEdxVspHypoNeg"][idEdxType]->SetBinContent(mMapHistCanvas["hdEdxVspHypoNeg"][idEdxType]->GetXaxis()->FindBin(pTPC), mMapHistCanvas["hdEdxVspHypoNeg"][idEdxType]->GetYaxis()->FindBin(dEdxTot[idEdxType]), Hypothesis);
+        }
```

Comment:
I think this can be simplified:
```suggestion
        auto pidHist = mMapHistCanvas[(track.getCharge() > 0) ? "hdEdxVspHypoPos" : "hdEdxVspHypoNeg"][idEdxType];
          pidHist->SetBinContent(pidHist->GetXaxis()->FindBin(pTPC), pidHist->GetYaxis()->FindBin(dEdxTot[idEdxType]), pidHypothesis + 1);
```

Reply:
This does not work, unfortunately, since the histograms in the map are unique_ptr. Either I can make them non-unique ptr or we go with the old method

Reply:
Can you try
```suggestion
 auto pidHist = mMapHistCanvas[(track.getCharge() > 0) ? "hdEdxVspHypoPos" : "hdEdxVspHypoNeg"][idEdxType].get();
          pidHist->SetBinContent(pidHist->GetXaxis()->FindBin(pTPC), pidHist->GetYaxis()->FindBin(dEdxTot[idEdxType]), pidHypothesis + 1);
```